integrations/langchain/src/databricks_langchain/vector_search_retriever_tool.py

```diff
@@ -0,0 +1,81 @@
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, Field, model_validator, PrivateAttr
+
+from databricks_langchain import DatabricksVectorSearch
+from databricks_langchain.utils import IndexDetails
+from langchain_core.embeddings import Embeddings
+from langchain_core.tools import BaseTool
+
+
+class VectorSearchRetrieverTool(BaseTool):
+    """
+    A utility class to create a vector search-based retrieval tool for querying indexed embeddings.
+    This class integrates with a Databricks Vector Search and provides a convenient interface
+    for building a retriever tool for agents.
+    """
+
+    index_name: str = Field(..., description="The name of the index to use, format: 'catalog.schema.index'.")
+    num_results: int = Field(10, description="The number of results to return.")
+    columns: Optional[List[str]] = Field(None, description="Columns to return when doing the search.")
+    filters: Optional[Dict[str, Any]] = Field(None, description="Filters to apply to the search.")
+    query_type: str = Field("ANN", description="The type of query to run.")
+    tool_name: Optional[str] = Field(None, description="The name of the retrieval tool.")
+    tool_description: Optional[str] = Field(None, description="A description of the tool.")
+    # TODO: Confirm if we can add these two to the API to support direct-access indexes or a delta-sync indexes with self-managed embeddings,
+    text_column: Optional[str] = Field(None, description="If using a direct-access index or delta-sync index, specify the text column.")
+    embedding: Optional[Embeddings] = Field(None, description="Embedding model for self-managed embeddings.")
```

Comment:
These two fields are required for [direct-access indexes or delta-sync indexes with self-managed embeddings](https://github.com/databricks/databricks-ai-bridge/blob/main/integrations/langchain/src/databricks_langchain/vectorstores.py#L108-L119). Should we support these additional fields?

Reply:
I feel like if we support it for DatabricksVectorSearch it makes sense to support it here.

Reply:
Yeah, seems reasonable to support these, though I'd say it's worth asking vector search folks how commonly direct access indexes are used, if it's infrequent we could drop this to start with to simplify the API/testing surface

Reply:
No need to block this PR on that though, I figure we'll need this eventually anyways, would just be good for us to know

Reply:
[Linking the slack thread I started here](https://databricks.slack.com/archives/C05BKRLUMC4/p1734703211820509)